docs/core/development/policy/coding-standard.rst

```diff
@@ -625,6 +625,126 @@ Any attribute, method argument, or method name that corresponds *directly* to a
+* For security features you might documented the emitted logs and then they will became part of the contract.
+
+Example of usage:
+
+.. code-block:: python
+
+    from twisted.logger import Logger
+
+    class SomeStuff(object):
+        """
+        Class using the new logging system as an instance member
+        """
+
+        _log = Logger()
+
+        def getUser(self, name):
+            try:
+                return self._otherStuff()
+            except Exception as error:
+                self._log.info(
+                    "Unable to check {user!s}", user=name, error=error)
+
+                raise UnauthorizedLogin()
+
+
+General testing rules:
+
+* New code calling the new logging system will test that the log is emitted.
+* You don't need to explicitly check all members of the structured log.
+* Emitted logs which are documented should have detailed tests which check all the conditions described in the documentation.
+
+
+Example of testing the code:
+
+.. code-block:: python
+
+    def test_getUserNonexistentDatabase(self):
+        """
+        A missing db file will cause a permanent rejection of authorization
+        attempts.
+        """
+        self.db = checkers.FilePasswordDB('test_thisbetternoteverexist.db')
+
+        with self.captureLogs() as logs:
+            self.failUnlessRaises(
+            error.UnauthorizedLogin, self.db.getUser, 'some-user')
+
+        self.assertIn('Unable to check some-user', logs.asText)
+        events = logs.asEvents
+        self.assertEqual(1, len(events))
+        self.assertEqual(events[0]['user'], 'some-user')
+        self.assertIsInstance(events[0]['error'] , error.UnauthorizedLogin)
+        self.assertEqual(events['log_level'], logs.LogLevel.info)
+
+
+Working around `self._log` conflicts.
+For older code you might now be able to use `self._log` as it can be already used for other purpose.
+In this case you can just use a different name for the new logging object or
+use the :api:`twisted.logger._logger._loggerFor <_loggerFor>` helper.
+
+.. code-block:: python
+
+    from twisted.logger._logger import _loggerFor
+
+    class SomeStuff(object):
+        """
+        Class using the new logging system using the C{_loggerFor} helper as
+        C{_log} ivar was already used.
+        """
+
+        def __init__(self):
+            self._log = SomeOtherStuff()
+            # Use a different name for the new logging code.
+            self._logger = Logger()
+            self._logger.info(
+                "Initializing factory {factory!r}", factory=self)
+
+
+        def doSomething(self):
+            """
+            Use the C{_loggerFor} helper to emit the logs without conflicting
+            with C{_log}.
+            """
+            _loggerFor(self).info(
+                "Starting factory {factory!r}", factory=self)
+            self.otherStuff()
+
+
+To control the log output level, use the dedicated log level and filter the logs at runtime.
```

Comment:
An excellent point, although perhaps more germane to the documentation for the logging system itself?